protokube/pkg/protokube/etcd_manifest.go

```diff
@@ -227,6 +228,15 @@ func buildEtcdEnvironmentOptions(c *EtcdCluster) []v1.EnvVar {
 		}
 	}
 
+	// expose etcd metrics is requested
+	if c.MetricsPort != 0 {
+		options = append(options, v1.EnvVar{Name: "ETCD_LISTEN_METRICS_URLS", Value: fmt.Sprintf("%s://__address__:%d", scheme, c.MetricsPort)})
+	}
+
+	if notEmpty(c.MetricsType) {
```

Comment:
I think instead of checking the zero values of string and int, its more common in the Kops API to use pointers for optional fields and then checking for `nil`.

So your API types would be `*int` and `*string` and these if statements would be `if c.MetricsType != nil`

Reply:
snippet from few lines above:

```
	if notEmpty(c.TLSKey) {
		options = append(options, v1.EnvVar{Name: "ETCD_KEY_FILE", Value: c.TLSKey})
	}
```

where
```
	// TLSKey is the path to a client private key for etcd
	TLSKey string
```

Reply:
ah you're right. Its probably better to be consistent with the rest of `EtcdClusterSpec` 👍